internal/cache/file/downloader/job_test.go

```diff
@@ -780,3 +780,63 @@ func (dt *downloaderTest) Test_Invalidate_Download_Concurrent() {
 	defer dt.job.mu.Unlock()
 	AssertEq(nil, dt.job.removeJobCallback)
 }
+
+func (dt *downloaderTest) Test_validateChecksum_ForTamperedFileWhenEnableCrcCheckIsTrue() {
+	objectName := "path/in/gcs/file1.txt"
+	objectSize := 8 * util.MiB
+	objectContent := testutil.GenerateRandomBytes(objectSize)
+	dt.initJobTest(objectName, objectContent, DefaultSequentialReadSizeMb, uint64(2*objectSize), func() {})
+	// Start download
+	offset := int64(8 * util.MiB)
+	jobStatus, err := dt.job.Download(context.Background(), offset, true)
+	AssertEq(nil, err)
+	// Here the crc check will be successful
+	dt.waitForCrcCheckToBeCompleted()
+	AssertEq(Completed, dt.job.status.Name)
+	AssertEq(nil, dt.job.status.Err)
+	AssertGe(dt.job.status.Offset, offset)
+	// Verify file
+	dt.verifyFile(objectContent[:jobStatus.Offset])
+	// Verify fileInfoCache update
+	dt.verifyFileInfoEntry(uint64(jobStatus.Offset))
+	// Tamper the file
+	err = os.WriteFile(dt.fileSpec.Path, []byte("test"), 0644)
+	AssertEq(nil, err)
+
+	err = dt.job.validateChecksum()
+
+	AssertNe(nil, err)
+	AssertTrue(strings.Contains(err.Error(), "checksum mismatch detected"))
```

Comment:
We should add a check that the job.Status is FAILED, file in cache is deleted and the entry for file in file info cache is erased.

Reply:
This test is a bit different. The job downloads the file and then we are tampering the file and calling validate checksum to validate the CRC which fails. Setting job status to failed is being done by the caller if error is returned by validate checksum.

Reply:
done.